packages/snaps-controllers/src/interface/utils.ts

```diff
@@ -189,36 +407,45 @@ export function constructState(
     }
 
     // Stateful components inside a form
+    // TODO: This is becoming a bit of a mess, we should consider refactoring this.
```

Comment:
Probably time to use arrays and `.includes()`

Reply:
I tried but then we need to do some type tricks so it infers the correct props and component name down the line, since it was more refactor than just using `includes` I left a todo for another PR. I can add a ticket for this so someone can follow up

Reply:
Ah good point, yeah `includes` probably doesn't narrow the type. Paging @Mrtenz 

Definitely out of scope for this PR.

Reply:
https://github.com/MetaMask/snaps/issues/3235